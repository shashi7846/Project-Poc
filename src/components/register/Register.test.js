import { render, screen } from "@testing-library/react";
import { renderComponent } from "../../utils/test_helpers";
import Registrationform from "./form";

describe("howtoapply page", () => {
  test("intital render", () => {
    render(renderComponent(<Registrationform />));
    expect(screen.getAllByText(/Register/i)[0]).toBeInTheDocument();
  });
  test("Matching the snapshot of the Login", async () => {
    const { asFragment } = render(renderComponent(<Registrationform />));
    expect(asFragment()).toMatchSnapshot();
  });
});
