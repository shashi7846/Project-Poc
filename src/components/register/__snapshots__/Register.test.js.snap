// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`howtoapply page Matching the snapshot of the Login 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container mt-5"
    >
      <div
        class="d-flex justify-content-center h-100"
      >
        <div
          class="card bg-dark col-6"
        >
          <div
            class="card-header "
          >
            <h3
              class="login-name text-light"
            >
              Register
            </h3>
          </div>
          <div
            class="card-body"
            style="background-color: rgb(179, 0, 0);"
          >
            <form>
              <div
                class="form-check form-check-inline mt-3 text-light"
              >
                <h4>
                  <b>
                    Gender
                  </b>
                </h4>
              </div>
              <div
                class="Registergender form-check form-check-inline"
              >
                <input
                  aria-label="Male"
                  class="form-check-input"
                  id="male"
                  name="gender"
                  type="radio"
                  value="M"
                />
                <label
                  class="form-check-label text-light "
                >
                  Male
                </label>
              </div>
              <div
                class=" Registergender form-check form-check-inline text-light"
              >
                <input
                  class="form-check-input"
                  id="female"
                  name="gender"
                  type="radio"
                  value="F"
                />
                <label
                  class="form-check-label "
                >
                  Female
                </label>
              </div>
              <div
                class="input-group form-group"
              >
                <div
                  class="input-group-prepend"
                >
                  <span
                    class="input-group-text text-white"
                    style="min-width: 150px;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-user fa-beat-fade"
                      data-icon="user"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style="font-size: 1.75em; color: black;"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0S96 57.3 96 128s57.3 128 128 128zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
                        fill="currentColor"
                      />
                    </svg>
                    <label
                      class="text-dark ms-2"
                    >
                      <b>
                        UserName
                      </b>
                    </label>
                  </span>
                </div>
                <input
                  class="form-control ms-2"
                  id="username"
                  name="username"
                  placeholder="Username"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="input-group form-group mt-3"
              >
                <div
                  class="input-group-prepend"
                >
                  <span
                    class="input-group-text text-white"
                    style="min-width: 150px;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-envelopes-bulk fa-beat-fade"
                      data-icon="envelopes-bulk"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style="font-size: 1.75em; color: black;"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M96 0C78.3 0 64 14.3 64 32V224h96V192c0-35.3 28.7-64 64-64H448V32c0-17.7-14.3-32-32-32H96zM224 160c-17.7 0-32 14.3-32 32v32h96c35.3 0 64 28.7 64 64V416H544c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32H224zm240 64h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H464c-8.8 0-16-7.2-16-16V240c0-8.8 7.2-16 16-16zM32 256c-17.7 0-32 14.3-32 32v13L155.1 415.9c1.4 1 3.1 1.6 4.9 1.6s3.5-.6 4.9-1.6L320 301V288c0-17.7-14.3-32-32-32H32zm288 84.8L184 441.6c-6.9 5.1-15.3 7.9-24 7.9s-17-2.8-24-7.9L0 340.8V480c0 17.7 14.3 32 32 32H288c17.7 0 32-14.3 32-32V340.8z"
                        fill="currentColor"
                      />
                    </svg>
                    <label
                      class="text-dark ms-2"
                    >
                      <b>
                        Email ID
                      </b>
                    </label>
                  </span>
                </div>
                <input
                  aria-label="email"
                  class="form-control ms-2"
                  id="email"
                  name="email"
                  placeholder="email"
                  type="email"
                  value=""
                />
              </div>
              <div
                class="input-group form-group mt-3"
              >
                <div
                  class="input-group-prepend"
                >
                  <span
                    class="input-group-text text-white "
                    style="min-width: 150px;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-calendar fa-beat-fade"
                      data-icon="calendar"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style="font-size: 1.75em; color: black;"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
                        fill="currentColor"
                      />
                    </svg>
                    <label
                      class="text-dark ms-2"
                    >
                      <b>
                        DateOfBirth
                      </b>
                    </label>
                  </span>
                </div>
                <input
                  class="form-control ms-2"
                  id="date"
                  name="date"
                  placeholder="date of birth"
                  type="date"
                  value=""
                />
              </div>
              <div
                class="input-group form-group mt-3"
              >
                <div
                  class="input-group-prepend"
                >
                  <span
                    class="input-group-text text-white"
                    style="min-width: 150px;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-key fa-beat-fade"
                      data-icon="key"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style="font-size: 1.75em; color: black;"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M336 352c97.2 0 176-78.8 176-176S433.2 0 336 0S160 78.8 160 176c0 18.7 2.9 36.8 8.3 53.7L7 391c-4.5 4.5-7 10.6-7 17v80c0 13.3 10.7 24 24 24h80c13.3 0 24-10.7 24-24V448h40c13.3 0 24-10.7 24-24V384h40c6.4 0 12.5-2.5 17-7l33.3-33.3c16.9 5.4 35 8.3 53.7 8.3zm40-176c-22.1 0-40-17.9-40-40s17.9-40 40-40s40 17.9 40 40s-17.9 40-40 40z"
                        fill="currentColor"
                      />
                    </svg>
                    <label
                      class="text-dark ms-2"
                    >
                      <b>
                        Password
                      </b>
                    </label>
                  </span>
                </div>
                <input
                  class="form-control ms-2 "
                  id="password"
                  name="password"
                  placeholder="set password"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="input-group form-group mt-3"
              >
                <div
                  class="input-group-prepend"
                >
                  <span
                    class="input-group-text text-white "
                    style="min-width: 150px;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-key fa-beat-fade"
                      data-icon="key"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style="font-size: 1.75em; color: black;"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M336 352c97.2 0 176-78.8 176-176S433.2 0 336 0S160 78.8 160 176c0 18.7 2.9 36.8 8.3 53.7L7 391c-4.5 4.5-7 10.6-7 17v80c0 13.3 10.7 24 24 24h80c13.3 0 24-10.7 24-24V448h40c13.3 0 24-10.7 24-24V384h40c6.4 0 12.5-2.5 17-7l33.3-33.3c16.9 5.4 35 8.3 53.7 8.3zm40-176c-22.1 0-40-17.9-40-40s17.9-40 40-40s40 17.9 40 40s-17.9 40-40 40z"
                        fill="currentColor"
                      />
                    </svg>
                    <label
                      class="text-dark ms-2"
                    >
                      <b>
                        Confirm
                      </b>
                    </label>
                  </span>
                </div>
                <input
                  class="form-control ms-2"
                  id="confirmPassword"
                  name="confirmPassword"
                  placeholder="confirm password"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="input-group form-group mt-3"
              >
                <div
                  class="input-group-prepend"
                >
                  <span
                    class="input-group-text text-white"
                    style="min-width: 150px;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-phone fa-beat-fade"
                      data-icon="phone"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style="font-size: 1.75em; color: black;"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M164.9 24.6c-7.7-18.6-28-28.5-47.4-23.2l-88 24C12.1 30.2 0 46 0 64C0 311.4 200.6 512 448 512c18 0 33.8-12.1 38.6-29.5l24-88c5.3-19.4-4.6-39.7-23.2-47.4l-96-40c-16.3-6.8-35.2-2.1-46.3 11.6L304.7 368C234.3 334.7 177.3 277.7 144 207.3L193.3 167c13.7-11.2 18.4-30 11.6-46.3l-40-96z"
                        fill="currentColor"
                      />
                    </svg>
                    <label
                      class="text-dark ms-2"
                    >
                      <b>
                        MobileNo.
                      </b>
                    </label>
                  </span>
                </div>
                <input
                  class="form-control ms-2"
                  id="phone"
                  name="phone"
                  placeholder="Enter Mobile Number"
                  type="tel"
                  value=""
                />
              </div>
              <div
                class="form-group mt-2"
              >
                <input
                  aria-label="Register"
                  class="btn float-right  btn-dark mt-1 px-4"
                  type="submit"
                  value="Register"
                />
              </div>
            </form>
          </div>
          <div
            class="card-footer"
          >
            <div
              class="d-flex justify-content-center links text-light"
            >
              Already have an account?
              <a
                href="/login"
              >
                Login
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
